tests/test_text.py

```diff
@@ -0,0 +1,223 @@
+# -*- coding: utf-8 -*-
+# Copyright FMR LLC <opensource@fidelity.com>
+# SPDX-License-Identifier: GNU GPLv3
+import pandas as pd
+
+from feature.selector import Selective, SelectionMethod
+from test_base import BaseTest
+from textwiser import TextWiser, Embedding, PoolOptions, Transformation, WordOptions
+
+
+
+class TestText(BaseTest):
+
+    """
+        optimization_method : random
+        num_features:
+            - t = maximum number of features defines by user. The cost_metric input argument should ignore
+            - t = None: the number of feature computed by solving a set cover problem with cost metrics
+                (unicost or diverse)
+    """
+    def test_text_based_random_max_num_feature(self):
+        data = pd.DataFrame(
+            {"item1": ["this is a sentences with more common words and more words to increase frequency"],
+             "item2": ["second one in list of more frequent sentences with some repeated words"],
+             "item3": ["a word for more complexity and longer length"],
+             "item4": ["sentence with a lot of repeated common words and more words to increase frequency"],
+             "item5": ["more frequent words with a valid vector and more words to increase frequency"]})
+        labels = pd.DataFrame({"item1": [0, 1, 0, 0, 0], "item2": [1, 0, 0, 1, 0], "item3": [0, 0, 1, 0, 1],
+                               "item4": [0, 1, 1, 0, 1], "item5": [0, 1, 0, 0, 1]})
+
+        print(data)
```

Comment:
I assume these are for debugging for now.. in general, when runnign the code we should not get any prints

Reply:
same comment in other tests.

Reply:
Yes, I added them for debugging purposes. I removed the print(data) and print(labels). Line selector.transform(data), the transform used only for printing selected features since decision makes in fit. Any tips on what the best functionality is for the transform function?

Reply:
That is a GREAT question! @takojunior  @bkleyn  for visibility. 

We talked about different Text_Based methods, which are essentially captured in the rows of your Slide, but did not talk about what fit() and transform() means for each method. 

That's a bit of an open ended design decision. 

For example, 
-  In one extreme, we can say transform() does nothing and fit() implements the methods in its entirety. 
- In the other extreme, transform() does all the job and fit just returns the selected columns. 

A hybrid might be possible; that is; the work for the method is split across Transform() and Fit(). The guiding principle being, transform does the necessary job/transformation for the method to work, and fit() does the main bulk of the work. 

For example, for random() if a K is given, transform() does not need to anything and fit() return K random columns. IF a K is not given, then transform() can figure out what the K is usign unicost or diverse as desired. Essentially, it "prepares" what's all needed to make the final selection. Then fit() again returns k random columns. 

Whatever "design decision" we make here, we NEED to be consistent across all rows in the Slide that the idea behind the Transform() and Fit() logic is the same. So we might ne able to decide it solely based on looking at Random() 

Thoughts from others?

Reply:
I feel the hybrid approach is reasonable. But I try to have one step back. For random(), and other methods, we are supposed to run fit() first and then transform() right? Or are we allow user to call transform() directly? I feel that if fit() already solves the selection by returning k columns (when k is not set, it can be figured out by solving the set_cover problem), then the transform() might just use the k returned indices to extract the data?

Reply:
You'r right.. in my comment I made a mistake in the order of FIT + TRANSFORM. Let me edit my comment to avoid confusion

Reply:
Comment updated, it should follow FIT + TRANSFORM sequence now.

Reply:
In the current implementation, the fit() function selects the desired number of text-based features using a content selection method and stores the selected feature names in the selected_features_ attribute. Then, the transform() function extracts these selected features from the input data and prints them to the console before returning them as a pandas DataFrame. This approach ensures that the selected features are consistently used for both training and inference.
Shall we continue with this approach?

Reply:
Sounds reasonable to me.. Let's proceed with as-is now.. and have a mental note for it ONCE we are done with all the methods and then we can see if FIT() + TRANSFORM() operations are somewhat meaningful/consistent across methods.